File: frontend/src/app/services/docente.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Docente } from '../models/docente.model'; // Asegúrate de tener este modelo definido

@Injectable({
  providedIn: 'root',
})
export class DocenteService {
  private apiUrl = 'http://localhost:8080/docente'; // URL de tu API

  constructor(private http: HttpClient) {}

  // Obtener todos los docentes
  obtenerTodosLosDocentes(): Observable<Docente[]> {
    return this.http.get<Docente[]>(this.apiUrl);
  }

  // Obtener un docente por ID
  obtenerDocentePorId(id: number): Observable<Docente> {
    return this.http.get<Docente>(`${this.apiUrl}/${id}`);
  }

  // Crear un nuevo docente
  crearDocente(docente: Docente): Observable<Docente> {
    return this.http.post<Docente>(this.apiUrl, docente);
  }

  // Actualizar un docente existente
  actualizarDocente(id: number, docente: Docente): Observable<Docente> {
    return this.http.put<Docente>(`${this.apiUrl}/${id}`, docente);
  }

  // Eliminar un docente por ID
  eliminarDocente(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}

File: backend/src/main/java/com/example/gestion_academica/service/DocenteService.java
package com.example.gestion_academica.service;

import java.util.List;
import java.util.Optional;

import org.springframework.http.ResponseEntity;


import com.example.gestion_academica.entity.Docente;


public interface DocenteService {
	List<Docente> obtenerTodosLosDocentes();

	Optional<Docente> obtenerDocentePorId(Long id);

	ResponseEntity<Docente> crearDocente(Docente docente);

	boolean eliminarDocente(Long id);

	ResponseEntity<Docente> actualizarDocente(Long id, Docente updatedDocente);
}
